lottie-compose/src/main/java/com/airbnb/lottie/compose/animateLottieComposition.kt

```diff
@@ -0,0 +1,72 @@
+package com.airbnb.lottie.compose
+
+import androidx.compose.runtime.*
+import com.airbnb.lottie.LottieComposition
+
+/**
+ * Returns a [LottieAnimatable] representing the progress of an animation.
+ *
+ * Because the state is mutable, you can modify its value and the internal animation
+ * will continue animating from the value you set. The progress will snap to the value you
+ * set without changing the repeat count.
+ *
+ * There is also a suspending version of this that takes progress as a MutableState<Float>
+ * as a required second parameter.
+ *
+ * You do not have to use this to animate a Lottie composition. You may create your own animation
+ * and pass its progress to [LottieComposition].
+ *
+ * @param composition The composition to render. This should be retrieved with [lottieComposition].
+ * @param isPlaying Whether or not the animation is currently playing. Note that the internal
+ *                  animation may end due to reaching the target iterations count. If that happens,
+ *                  the animation may stop even if this is still true. You may want to use
+ *                  onFinished to set isPlaying to false but in many cases, it won't matter.
+ * @param restartOnPlay If isPlaying switches from false to true, restartOnPlay determines whether
+ *                      the progress and iteration gets reset.
+ * @param clipSpec A [LottieClipSpec] that specifies the bound the animation playback
+ *                 should be clipped to.
+ * @param speed The speed the animation should play at. Numbers larger than one will speed it up.
+ *              Numbers between 0 and 1 will slow it down. Numbers less than 0 will play it backwards.
+ * @param iterations The number of times the animation should repeat before stopping. It must be
+ *                    a positive number. [LottieConstants.IterateForever] can be used to repeat forever.
+ */
+@Composable
+fun animateLottieComposition(
```

Comment:
What's the use case for this, or is it more of an impl detail? :-)

Reply:
@doris4lt It's effectively the declarative version of LottieAnimatable

Reply:
When would a user use it?

Reply:
@doris4lt I think _most_ of the time. 90%+ maybe? The imperative version (LottieAnimatable) should be relatively rare IMO. Maybe this should be called `animateLottieCompisitionAsState` to match Compose `animate*asState`. Same analogy here.

Reply:
When should devs use this:

```
@Composable
fun LottieAnimation(
    composition: LottieComposition?,
    modifier: Modifier = Modifier,
    isPlaying: Boolean = true,
    restartOnPlay: Boolean = true,
    clipSpec: LottieClipSpec? = null,
    speed: Float = 1f,
    iterations: Int = 1,
    imageAssetsFolder: String? = null,
    imageAssetDelegate: ImageAssetDelegate? = null,
    outlineMasksAndMattes: Boolean = false,
    applyOpacityToLayers: Boolean = false,
    enableMergePaths: Boolean = false,
) 
```
over `animateLottieComposition` + `LottieAnimation`?

If the difference between the two is just whether progress is observable, it may be worth merging the two options above by making the progress always observable and externally immutable, which can be achieved by returning LottieAnimationState.

The benefit of doing this is to reduce selection anxiety when devs are faced with a number of API options. :P

Reply:
It's partly that and partly that one is declarative and one is imperative. I like having the two options because the declarative version doesn't require 2 setup lines of code. You can handle the entire thing in just composable parameters.